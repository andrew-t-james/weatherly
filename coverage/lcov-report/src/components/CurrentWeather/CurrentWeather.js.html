<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/CurrentWeather/CurrentWeather.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/components/CurrentWeather</a> CurrentWeather.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import './CurrentWeather.css';
&nbsp;
const CurrentWeather = props =&gt; {
  const { city, icon, temperature, weather, high, low, description } = props.currentWeather;
&nbsp;
  return (
    &lt;section className="current-weather"&gt;
      &lt;h2
        tabIndex="0"
        className="current-weather__title"
        aria-label={city}&gt;
        {city}
      &lt;/h2&gt;
      &lt;p
        tabIndex="0"
        className="current-weather__weather"
        aria-label={weather}&gt;
        {weather}
      &lt;/p&gt;
      &lt;p
        tabIndex="0"
        className="current-weather__temperature"
        aria-label={temperature}
      &gt;{temperature} &amp;#176;
      &lt;/p&gt;
      &lt;i
        tabIndex="0"
        className={`current-weather__icon wi wi-wu-${icon}`}
        aria-label={icon}
        alt={icon}/&gt;
      {
        high &amp;&amp;
        &lt;p
          tabIndex="0"
          aria-label={`The high will be ${high} today`}
          className="current-weather__high"&gt;
          {high}&amp;#176; &amp;uarr;
        &lt;/p&gt;
      }
      {
        low &amp;&amp;
        &lt;p
          tabIndex="0"
          aria-label={`The Low will be ${low} today`}
          className="current-weather__low"&gt;
          {low}&amp;#176; &amp;darr;
        &lt;/p&gt;
      }
      {
        description &amp;&amp;
        &lt;p
          tabIndex="0"
          aria-label={description}
          className="current-weather__description"&gt;
          {description}
        &lt;/p&gt;
      }
    &lt;/section&gt;
  );
};
&nbsp;
CurrentWeather.propTypes = {
  currentWeather: PropTypes.object,
  city: PropTypes.string,
  temperature: PropTypes.number,
  high: PropTypes.string,
  low: PropTypes.string,
  icon: PropTypes.string,
  weather: PropTypes.string,
  description: PropTypes.string
};
&nbsp;
export default CurrentWeather;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu May 31 2018 09:58:02 GMT-0600 (MDT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
