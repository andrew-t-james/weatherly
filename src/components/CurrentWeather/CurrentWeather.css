.current-weather {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 30rem;
  justify-content: center;;
}

.current-weather__title {
  font-size: 3rem;
}

.current-weather__weather,
.current-weather__temperature,
.current-weather__description {
  font-size: 2rem;
}

.current-weather__temperature {
  margin-bottom: 2rem;
}

.current-weather__icon {
  font-size: 8rem;
}

.current-weather__high,
.current-weather__low {
  font-size: 1.8rem;
}

.current-weather__description {
  margin-top: 1rem;
}