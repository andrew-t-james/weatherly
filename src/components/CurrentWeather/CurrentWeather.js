import React from 'react';

import './CurrentWeather.css';

const CurrentWeather = (props) => {
  const { city, icon, iconUrl, temperature, weather, high, low, description } = props.currentWeather;

  return (
    <div className="current-weather">
      <h2 className="current-weather__title">{city}</h2>
      <p className="current-weather__weather">{weather}</p>
      <p className="current-weather__temperature">{temperature}</p>
      <img className="current-weather__icon" src={iconUrl} alt={icon}/>
      <div className="high-low">
        <p className="current-weather__high">{high} &uarr;</p>
        <p className="current-weather__low">{low} &darr;</p>
      </div>
      <p className="current-weather__description">{description}</p>
    </div>
  );
};

export default CurrentWeather;
