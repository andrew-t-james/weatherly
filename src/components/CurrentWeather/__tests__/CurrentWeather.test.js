import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import CurrentWeather from '../CurrentWeather.js';


describe('Current Weather unit tests', () => {
  let currentWeather;

  const currentLocation = {
    city: 'Louisville',
    description: 'Sun and clouds mixed. High 51F. Winds NE at 10 to 15 mph.',
    high: '51',
    icon: 'mostlycloudy',
    iconUrl: 'http://icons.wxug.com/i/c/k/mostlycloudy.gif',
    low: '32',
    temperature: 46,
    weather: 'Mostly Cloudy',
    weekDay: 'Wednesday'
  };

  beforeEach(() => currentWeather = shallow(<CurrentWeather currentWeather={currentLocation}/>));

  test('It should display data correctly', () => {
    const expectedTitle = 'Louisville';
    const actualTitle = currentWeather.find('h2').text();

    const expectedWeather = 'Mostly Cloudy';
    const actualWeather = currentWeather.find('.current-weather__weather').text();

    const expectedTemperature = '46';
    const actualTemperature = currentWeather.find('.current-weather__temperature').text();

    const expectedHigh = '51° ↑ High ';
    const actualHigh = currentWeather.find('.current-weather__high').text();

    const expectedLow = '32° ↓ Low';
    const actualLow = currentWeather.find('.current-weather__low').text();

    const expectedDescription = 'Sun and clouds mixed. High 51F. Winds NE at 10 to 15 mph.';
    const actualDescription = currentWeather.find('.current-weather__description').text();

    expect(actualTitle).toBe(expectedTitle);
    expect(expectedWeather).toBe(actualWeather);
    expect(actualTemperature).toBe(expectedTemperature);
    expect(actualHigh).toBe(expectedHigh);
    expect(actualLow).toBe(expectedLow);
    expect(actualDescription).toBe(expectedDescription);
  });

  test('it should match snapshot', () => {
    const currWeather = renderer
      .create(<CurrentWeather currentWeather={currentLocation} />)
      .toJSON();

    expect(currWeather).toMatchSnapshot();
  });
});