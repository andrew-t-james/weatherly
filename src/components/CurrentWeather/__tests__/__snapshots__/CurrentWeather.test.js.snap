// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Current Weather unit tests it should match snapshot 1`] = `
<section
  className="current-weather"
>
  <h2
    className="current-weather__title"
  >
    Louisville
  </h2>
  <p
    className="current-weather__weather"
  >
    Mostly Cloudy
  </p>
  <p
    className="current-weather__temperature"
  >
    46
     °
  </p>
  <i
    alt="mostlycloudy"
    className="current-weather__icon wi wi-wu-mostlycloudy"
  />
  <p
    className="current-weather__high"
  >
    51
    ° ↑
  </p>
  <p
    className="current-weather__low"
  >
    32
    ° ↓
  </p>
  <p
    className="current-weather__description"
  >
    Sun and clouds mixed. High 51F. Winds NE at 10 to 15 mph.
  </p>
</section>
`;
